app/code/Magento/NewRelicReporting/Plugin/StatePlugin.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\NewRelicReporting\Plugin;
+
+use Magento\Framework\App\State;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\NewRelicReporting\Model\Config;
+use Magento\NewRelicReporting\Model\NewRelicWrapper;
+use Psr\Log\LoggerInterface;
+
+class StatePlugin
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var NewRelicWrapper
+     */
+    private $newRelicWrapper;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param Config $config
+     * @param NewRelicWrapper $newRelicWrapper
+     */
+    public function __construct(
+        Config $config,
+        NewRelicWrapper $newRelicWrapper,
+        LoggerInterface $logger
+    ) {
+        $this->config = $config;
+        $this->newRelicWrapper = $newRelicWrapper;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Set separate appname
+     *
+     * @param State $subject
+     * @param null $result
+     * @return void
```

Comment:
your code returns some data, please update docblock

Reply:
Also wouldn't be wrong to auto-format this so that the variable names are aligned under each other. I'm not sure if this is required in the magento coding standards though...

Reply:
@igortregub  we can use "mixed", but actually it doesn't matter. Will update PR
@abbasalis it's not required in magento coding standards